jupyter_sphinx/execute.py

```diff
@@ -257,7 +257,15 @@ def write_notebook_output(notebook, output_dir, notebook_name):
         os.path.join(output_dir, notebook_name + ".ipynb"),
     )
     # Write a script too.
-    ext = notebook.metadata.language_info.file_extension
+    if ntbk.metadata.get("language_info", {}).get("file_extension", None) is None:
+        ext = ".txt"
+        js.logger.warning(
+            "Notebook code has no file extension metadata, " "defaulting to `.txt`",
+            # TODO correct location
+            # location=document.settings.env.docname,
+        )
+    else:
+        ext = notebook.metadata.language_info.file_extension
```

Comment:
```suggestion
    ext = notebook.metadata.language_info.file_extension
    if ext is None:
        ext = ".txt"
        js.logger.warning(
            "Notebook code has no file extension metadata, " "defaulting to `.txt`",
            # TODO correct location
            # location=document.settings.env.docname,
        )
```

Reply:
`notebook.metadata.language_info.file_extension` doesn't return None if missing though; it raises an `AttributeError` (See executablebooks/MyST-NB#152)

Reply:
Sorry, fixed. My multitasking isn't superb.